crates/bevy_animation/src/keyframes.rs

```diff
@@ -0,0 +1,571 @@
+///     # use bevy_animation::{VariableCurve, prelude::TranslationKeyframes};
+///     # use bevy_core::Name;
+///     # use bevy_math::Vec3;
+///     # let animation_target_id = AnimationTargetId::from(&Name::new("Test"));
+///     let mut animation_clip = AnimationClip::default();
+///     animation_clip.add_curve_to_target(
+///         animation_target_id,
+///         VariableCurve::linear::<TranslationKeyframes>(
+///             vec![0.0, 1.0],
+///             vec![Vec3::ZERO, Vec3::ONE],
+///         ),
+///     );
+#[derive(Clone, Reflect, Debug, Deref, DerefMut)]
+pub struct TranslationKeyframes(pub Vec<Vec3>);
+
+/// Keyframes for animating [`Transform::scale`].
+///
+/// An example of a [`crate::AnimationClip`] that animates translation:
+///
+///     # use bevy_animation::{AnimationClip, AnimationTargetId, Interpolation};
+///     # use bevy_animation::{VariableCurve, prelude::ScaleKeyframes};
+///     # use bevy_core::Name;
+///     # use bevy_math::Vec3;
+///     # let animation_target_id = AnimationTargetId::from(&Name::new("Test"));
+///     let mut animation_clip = AnimationClip::default();
+///     animation_clip.add_curve_to_target(
+///         animation_target_id,
+///         VariableCurve::linear::<ScaleKeyframes>(
+///             vec![0.0, 1.0],
+///             vec![Vec3::ONE, Vec3::splat(2.0)],
+///         ),
+///     );
+#[derive(Clone, Reflect, Debug, Deref, DerefMut)]
+pub struct ScaleKeyframes(pub Vec<Vec3>);
+
+/// Keyframes for animating [`Transform::rotation`].
+///
+/// An example of a [`crate::AnimationClip`] that animates translation:
+///
+///     # use bevy_animation::{AnimationClip, AnimationTargetId, Interpolation};
+///     # use bevy_animation::{VariableCurve, prelude::RotationKeyframes};
+///     # use bevy_core::Name;
+///     # use bevy_math::Quat;
+///     # use std::f32::consts::FRAC_PI_2;
+///     # let animation_target_id = AnimationTargetId::from(&Name::new("Test"));
+///     let mut animation_clip = AnimationClip::default();
+///     animation_clip.add_curve_to_target(
+///         animation_target_id,
+///         VariableCurve::linear::<RotationKeyframes>(
+///             vec![0.0, 1.0],
+///             vec![
+///                 Quat::from_rotation_x(FRAC_PI_2),
+///                 Quat::from_rotation_y(FRAC_PI_2),
+///             ],
+///         ),
+///     );
+#[derive(Clone, Reflect, Debug, Deref, DerefMut)]
+pub struct RotationKeyframes(pub Vec<Quat>);
+
+/// Keyframes for animating [`MorphWeights`].
+#[derive(Clone, Debug, Reflect)]
+pub struct MorphWeightsKeyframes {
+    /// The total number of morph weights.
+    pub morph_target_count: usize,
+
+    /// The morph weights.
+    ///
+    /// The length of this vector should be the total number of morph weights
+    /// times the number of keyframes.
+    pub weights: Vec<f32>,
+}
+
+impl From<Vec<Vec3>> for TranslationKeyframes {
+    fn from(value: Vec<Vec3>) -> Self {
+        Self(value)
+    }
+}
```

Comment:
Similar to my other suggestion, changing this would allow calling the `VariableCurve` constructors using arrays.

```suggestion
impl<T: Into<Vec<Vec3>>> From<T> for TranslationKeyframes {
    fn from(value: T) -> Self {
        Self(value.into())
    }
}
```

Reply:
Ok, I made that change.